### PR TITLE
Parameter query and whereClause both specified

### PR DESCRIPTION
**Pull request by vzubriichuk**
Allow to use query + where

### CODE REVIEW

File: src/main/scala/palefat/data/mirror/Config.scala [hunk 1]
```diff
@@ -150,8 +145,8 @@ case class Config(
   }
 
   val query: String =
-    if (_query.isEmpty) {
-      SqlBuilder.buildSelectTableSQL(schema, tab, whereClause.toString)
+    if (_query.isEmpty || (_query.nonEmpty && _whereClause.nonEmpty)) {
```
Comment (hanna-liashchuk) [inline src/main/scala/palefat/data/mirror/Config.scala]: this won't work cause if `query` is empty you get true in the first part and the second one is ignored

Comment (hanna-liashchuk) [reply src/main/scala/palefat/data/mirror/Config.scala]: I suggest 
  ```
val query: String =
    if (_query.isEmpty) {
      SqlBuilder.buildSelectTableSQL(schema, tab, whereClause.toString)
    } else {
      if (_whereClause.nonEmpty) {
        s"(${_query} where ${whereClause.toString}) as subq"
      } else {
        s"(${_query}) as subq"
      }
    }
```

and no need to modify the method in this case

**Review by hanna-liashchuk — CHANGES_REQUESTED** [review]

**Review by hanna-liashchuk — DISMISSED** [review]

File: src/main/scala/palefat/data/mirror/Config.scala [hunk 2]
```diff
@@ -152,7 +147,13 @@ case class Config(
   val query: String =
     if (_query.isEmpty) {
       SqlBuilder.buildSelectTableSQL(schema, tab, whereClause.toString)
-    } else { s"(${_query}) as subq" }
+    } else {
+      if (_whereClause.nonEmpty) {
```
Comment (VShkaberda) [inline src/main/scala/palefat/data/mirror/Config.scala]: `else if` or make  a new method `SqlBuilder.buildQuerySQL`.

**Review by VShkaberda — CHANGES_REQUESTED** [review]
Now it is forbidden to use `where` condition in `query` if `where` argument is given. It should be added to Readme.

File: src/main/scala/palefat/data/mirror/Config.scala (hunk 1 above)
Comment (vzubriichuk) [reply src/main/scala/palefat/data/mirror/Config.scala]: Done

**Review by VShkaberda — APPROVED** [review]

**Review by hanna-liashchuk — APPROVED** [review]

**Pull request merged**